.github/workflows/nightly-python-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Nightly Python test
+on:
+  schedule:
+    - cron: '0 1 * * *' # Always run at 1:00 a.m.
+  workflow_dispatch:
+
+jobs:
+  full-python-test:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3.3.0
+        with:
+          fetch-depth: 0
+
+      - name: Cache Giskard test resources
+        uses: actions/cache@v3
+        with:
+          path: ~/.giskard
+          key: ${{ runner.os }}-giskard-test-resources-${{ hashFiles('python-client/tests/fixtures/**/*py')}}
+          restore-keys: ${{ runner.os }}-giskard-test-resources
+
+      - name: Cache Python dependencies
+        uses: actions/cache@v3
+        with:
+          path: python-client/.venv
+          key: ${{ runner.os }}-client-${{ hashFiles('python-client/pdm.lock')}}
+          restore-keys: ${{ runner.os }}-client
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Gradle # To cache ~/.gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Get number of available cores for Python test workers
+        id: nb_workers
+        run: python -c 'import os; print(f"nb_workers={max(1,os.cpu_count()-1)}")' >> $GITHUB_OUTPUT
+
+      - name: Run Python commands
+        run: |
+          python -m pip install virtualenv -U --upgrade pip
+          echo "Python ($(which python)) packages:\n $(python -m pip list)"
+
+      - name: Run full Python tests
```

Comment:
I'd actually do a full build (including all tests execution), not just python tests. Like this we'll know each morning that we've tested everything we could test